controllers/openstackdataplanedeployment_controller.go

```diff
@@ -0,0 +1,243 @@
+//+kubebuilder:rbac:groups=baremetal.openstack.org,resources=openstackbaremetalsets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;delete;
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete;
+//+kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;watch;create;update;patch;delete;
+//+kubebuilder:rbac:groups=k8s.cni.cncf.io,resources=network-attachment-definitions,verbs=get;list;watch
+//+kubebuilder:rbac:groups=network.openstack.org,resources=ipsets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=network.openstack.org,resources=ipsets/status,verbs=get
+//+kubebuilder:rbac:groups=network.openstack.org,resources=ipsets/finalizers,verbs=update
+//+kubebuilder:rbac:groups=network.openstack.org,resources=netconfigs,verbs=get;list;watch
+//+kubebuilder:rbac:groups=network.openstack.org,resources=dnsmasqs,verbs=get;list;watch
+//+kubebuilder:rbac:groups=network.openstack.org,resources=dnsmasqs/status,verbs=get
+//+kubebuilder:rbac:groups=network.openstack.org,resources=dnsdata,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=network.openstack.org,resources=dnsdata/status,verbs=get
+//+kubebuilder:rbac:groups=network.openstack.org,resources=dnsdata/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;delete;
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *OpenStackDataPlaneDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, _err error) {
+
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling Deployment")
+
+	// Fetch the OpenStackDataPlaneDeployment instance
+	instance := &dataplanev1.OpenStackDataPlaneDeployment{}
+	err := r.Client.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers. Return and don't requeue.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	helper, _ := helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		logger,
+	)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Always patch the instance status when exiting this function so we can persist any changes.
+	defer func() { // update the Ready condition based on the sub conditions
+		if instance.Status.Conditions.AllSubConditionIsTrue() {
+			instance.Status.Conditions.MarkTrue(
+				condition.ReadyCondition, condition.ReadyMessage)
```

Comment:
Should we also check the nodeSet status here as well?
https://github.com/openstack-k8s-operators/dataplane-operator/pull/380/files#diff-4430f5d9d58a002e0f54aa0013aecf8408c4850e263db289d25bf1eacdd7902b

```suggestion
		if instance.Status.Conditions.AllSubConditionIsTrue() || instanceNodeSet.IsReady() {
			instance.Status.Conditions.MarkTrue(
				condition.ReadyCondition, condition.ReadyMessage)
```

Reply:
Which would obviously also require that we retrieve the `instanceNodeSet` prior to this.

Reply:
It's not necessary actually since the DeploymentReady condition will handle this during the `for _, nodeSet := range nodeSets.Items ` loop later in this file.